src/framework/uicomponents/qml/MuseScore/UiComponents/NumberInputField.qml

```diff
@@ -0,0 +1,121 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+
+import MuseScore.Ui 1.0
+
+Item {
+    id: root
+
+    property int minValue: 0
+    property int maxValue: 999
+    property int value: 0
+
+    property alias font: textField.font
+
+    implicitWidth: textField.contentWidth
+    implicitHeight: 30
+
+    opacity: enabled ? 1 : ui.theme.itemOpacityDisabled
+
+    QtObject {
+        id: privateProperties
+
+        property int maxNumberLength: root.maxValue.toString().length
+
+        function pad(value) {
+            var str = value.toString()
+
+            while (str.length < maxNumberLength) {
+                str = "0" + str
+            }
+
+            return str
+        }
+    }
+
+    TextField {
```

Comment:
Isn't it better to use TextInputField here? 

It's already handling all the focus behaviours, key presses, etc.

Reply:
I do not think it's a good idea, because the behavior of this component is very different from TextInputField. Otherwise, we will overcomplicate the TextInputField implementation

Reply:
tell me briefly what is the difference